packages/flutter/lib/src/widgets/snapshot_widget.dart

```diff
@@ -0,0 +1,269 @@
+  /// snapshot will not be used and the child will be rendered as normal.
+  permissive,
+
+  /// the child is snapshotted, but only if all descendants can be snapshotted.
+  ///
+  /// An error is thrown if an unrasterizable view is encountered. This setting is
+  /// the default state of the [SnapshotWidgetController].
+  normal,
+
+  /// The child is snapshotted and any child platform views are ignored.
+  ///
+  /// This mode can be useful if there is a platform view descendant that does
+  /// not need to be included in the snapshot.
+  forced,
+}
+
+/// A controller for the [SnapshotWidget] that controls when the child image is displayed
+/// and when to regenerated the child image.
+///
+/// When the value of [enabled] is true, the [SnapshotWidget] will paint the child
+/// widgets based on the [SnapshotMode] of the snapshot widget.
+///
+/// To force [SnapshotWidget] to recreate the child image, call [clear].
+class SnapshotWidgetController extends ChangeNotifier {
+  /// Create a new [SnapshotWidgetController].
+  ///
+  /// By default, [enabled] is `false` and cannot be `null`.
+  SnapshotWidgetController({
+    bool enabled = false,
+  }) : _enabled = enabled;
+
+  /// Reset the snapshot held by any listening [SnapshotWidget].
+  ///
+  /// This has no effect if [enabled] is `false`.
+  void clear() {
+    notifyListeners();
+  }
+
+  /// Whether a snapshot of this child widget is painted in its place.
+  bool get enabled => _enabled;
+  bool _enabled;
+  set enabled(bool value) {
+    if (value == enabled) {
+      return;
+    }
+    _enabled = value;
+    notifyListeners();
+  }
+}
+
+/// A widget that replaces its child with a snapshoted version of the child.
+///
+/// This widget is useful for performing short animations that would otherwise
+/// be expensive or that cannot rely on raster caching. For example, scale and
+/// skew animations are often expensive to perform on complex children, as are
+/// blurs. For a short animation, a widget that contains these expensive effects
+/// can be replaced with a snapshot of it self and manipulated instead.
+///
+/// For example, the Android Q [ZoomPageTransitionsBuilder] uses a snapshot widget
+/// for the forward and entering route to avoid the expensive scale animation.
+/// This also has the effect of briefly pausing any animations on the page.
+///
+/// Generally, this widget should not be used in places where users expect the
+/// child widget to continue animation or to be responsive, such as an unbounded
+/// animation.
+///
+/// Caveats:
+///
+/// * The contents of platform views cannot be captured by a snapshot
+///   widget. If a platform view is encountered, then the snapshot widget will
+///   determine how to render its children based on the [SnapshotMode]. This
+///   defaults to [SnapshotMode.normal] which will throw an exception if a
+///   platform view is encountered.
+///
+/// * This widget is not supported on the HTML backend of Flutter for the web.
```

Comment:
document what "not supported" means. Is this widget always a no-op on the web? Are you never allowed to instantiate this on the web? What happens if you do?

Reply:
I arrived at this question because the ZoomTransition appears to unconditionally be using this widget. So, some modes of this widget must be supported on the web??

Reply:
Not sure the best way to phrase this, I tried some updates